README.md

```diff
@@ -122,7 +124,63 @@ If you want to deploy directly from the CLI, run `yarn vercel` and follow the st
 
 If you want to redeploy to the same production URL you can run `yarn vercel --prod`. If you omit the `--prod` flag it will deploy it to a preview/test URL.
 
-**Make sure your `packages/nextjs/scaffold.config.ts` file has the values you need.**
+**Make sure to check the values of your Scaffold Configuration before deploying your NextJS App.**
+
+### Scaffold Configuration
+
+You can configure different settings for your dapp at `packages/nextjs/scaffold.config.ts`.
+
+```ts
+export type ScaffoldConfig = {
+  targetNetwork: chains.Chain;
+  pollingInterval: number;
+  alchemyApiKey: string;
+  burnerWallet: {
+    enabled: boolean;
+    onlyLocal: boolean;
+  };
+  walletAutoConnect: boolean;
+};
+```
+
+The configuration parameters are described below, make sure to update the values according to your needs:
+
+- **targetNetwork**  
+  Sets the blockchain network where your dapp is deployed. Use values from `wagmi/chains`.
+
+- **pollingInterval**  
+  The interval in milliseconds at which your front-end application polls the RPC servers for fresh data. _Note that this setting does not affect the local network._
+
+- **alchemyApiKey**  
+  Default Alchemy API key from Scaffold ETH 2 for local testing purposes.  
+  It's recommended to obtain your own API key from the [Alchemy Dashboard](https://dashboard.alchemyapi.io/) and store it in an environment variable: `ALCHEMY_API_KEY` at `\packages\hardhat\.env` file.
+
+- **burnerWallet**  
+  You can activate the Burner Wallet feature, which provides a lightweight wallet for users.
+
+  - **enabled**  
+    Set it to false to completely remove the Burner Wallet from all networks.
+  - **onlyLocal**  
+    Set it to true to use the Burner Wallet only at local network (hardhat).
+
+- **walletAutoConnect**  
+  Set it to true to activate automatic wallet connection behavior:
+  - If the user was connected into a wallet before, on page reload it reconnects automatically.
+  - If user is not connected to any wallet, on reload, it connects to the burner wallet if `burnerWallet.enabled` is true and `burnerWallet.onlyLocal` is false.
+
+You can extend this configuration file, adding new parameters that you need to use across your dapp. For example:
```

Comment:
This would result in typescript errors without also modifying the `ScaffoldConfig` type, so not sure if we should encourage people to do this. Maybe a good discussion to have, to determine where custom config should reside.

Reply:
Or alternatively we could do:
```typescript
export type ScaffoldConfig = {
  targetNetwork: chains.Chain;
  pollingInterval: number;
  alchemyApiKey: string;
  burnerWallet: {
    enabled: boolean;
    onlyLocal: boolean;
  };
  walletAutoConnect: boolean;
} & Record<string, any>;
```

Reply:
Yep, it's a good discussion to have. The three options:

1. As Pablo was suggesting. They will need to update the types tho (not super big deal, since it's right on top)
2. `custom?: Record<string, any>;` as Samuel suggested. I like it.
3. `} & Record<string, any>;` second option that Samuel suggested. I don't like this one since any string will work (you can misspell any of the required config props and won't get an error).

I think it should be between 1 or 2.

1: If we see "ScaffoldConfig" as the _Custom App config_ that the user just created (e.g. similar to Header.tsx: it's a component from the _custom app_)
2: If we see "ScaffoldConfig" as an "immutable" (but extendable) config from the app (e.g. similar to Address.tsx: it's a component from _scaffold-eth_)

Not sure if I'm making sense :D

Reply:
Which one are we finally going with? I also prefer option 1

Reply:
Option 3. also gives error when you misspell one of the required props (but maybe a less obvious one).

I think Option 2. has the best of both worlds: it's noob-friendly, doesn't require meddling with the type above, and makes it clear throughout the codebase that a config prop is custom.

Reply:
I think that option 1 is good enough (for a beginner, learning that if he adds a new config item, he also needs to change the type definition, it is not a bad way of learning it), but option 2 sounds good too.